paddle/fluid/framework/parallel_executor.cc

```diff
@@ -252,7 +252,21 @@ ir::Graph *ParallelExecutorPrivate::ApplyMemoryOptimizePass(ir::Graph *graph) {
     VLOG(10) << "buffer_shared_inplace_pass Applied";
   }
 
-  if (build_strategy_.memory_optimize_) {
+  /**
```

Comment:
The comment logic doesn't match the code. In the comment you said 
"If gc is enabled, BuildStrategy.memory_optimize = False.
If gc is disabled, BuildStrategy.memory_optimize = True.
...
gc + memory_optimize is worse than gc only."

So your code logic should probably be without "if (!build_strategy_.memory_optimize_)" 
The if condition makes it possible for user to set both
build_strategy_.memory_optimize_ and gc to True.

Please consider to modify the comment or the code.

Reply:
Discussed offline. User can forcely turn on `memory_optimize` by setting True. The comment is right.